.github/workflows/rc_sync.yaml

```diff
@@ -0,0 +1,75 @@
+# This workflow will sync changes in the release candidate branch to main during a
+# feature freeze. The action is run daily and will only create a pull request when
+# there exists a release candidate branch corresponding to the current package version
+# as declared in catalyst/frontend/_version.py.
+
+name: Sync release candidate with main
+
+# Controls when the workflow will run
+on:
+  # Scheduled trigger every weekday at 2:47am UTC
+  schedule:
+  - cron:  '47 2 * * 1-5'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job to sync the main branch with changes from the rc
+  sync:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out the Catalyst repository
+      - uses: actions/checkout@v3
+
+      # Check for the existence of an rc branch for the current Catalyst version
+      - name: Check for rc branch
+        run: |
+          VERSION=$(python setup.py --version)
+          IFS=. read MAJ MIN PAT <<< "${VERSION%.dev0}"
+          RC_BRANCH="v${MAJ}.$((MIN-1)).${PAT}-rc0"
+          if git ls-remote --exit-code origin "refs/heads/$RC_BRANCH"; then
+            echo "branch_exists=true" >> $GITHUB_ENV
+            echo "rc_branch=$RC_BRANCH" >> $GITHUB_ENV
+          else
+            echo "branch_exists=false" >> $GITHUB_ENV
+          fi
+          echo "new_changes=false" >> $GITHUB_ENV
+
+      # Create a new branch at from the current rc to be used in the PR
+      - name: Create temp branch
+        if: ${{ env.branch_exists == 'true' }}
+        run: |
+          git fetch
+          git checkout "${{ env.rc_branch }}"
+          BRANCH="rc_$(date +'%Y-%m-%d-%H-%M-%S')"
+          echo "tmp_branch=$BRANCH" >> $GITHUB_ENV
+          git checkout -b "$BRANCH"
+          # Exclude some files known to be divergent during the release process from the PR
+          git checkout main -- catalyst/frontend/_version.py
```

Comment:
@dime10 I am just not sure there will not be a lot a merge issue with the changelog as we have a single file compared to PL that splits it for each release.

Reply:
Hmm yeah the changelog might throw conflicts every time 🤔

Reply:
@erick-xanadu did you need to resolve conflicts in the changelog when you have done rc syncs to main?

Reply:
@dime10 I did need to resolve conflicts in the changelog.

Reply:
If we want to fix that we need to split the changelog into separate release files

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1069 and https://github.com/PennyLaneAI/catalyst/pull/1070

Reply:
That should not be a blocker anymore